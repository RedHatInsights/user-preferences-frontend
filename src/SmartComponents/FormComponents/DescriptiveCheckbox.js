import React from 'react';
import { Checkbox } from '@patternfly/react-core';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import './descriptiveCheckbox.scss';

// eslint-disable-next-line no-unused-vars
const DescriptiveCheckbox = ({ name, label, description, isDanger, FieldProvider, formOptions, ...rest }) => (
    <Checkbox
        { ...rest }
        id={ `descriptive-checkbox-${name}` }
        className="pref-c__descriptive-checkbox"
        label={ <span className={ classNames('pref-c__checkbox-label', { 'pref-c__checkbox-label-error': isDanger }) }>{label}</span> }
        description={ <span className="pref-c__checkbox-description">{description}</span> }
    />
);

DescriptiveCheckbox.propTypes = {
    FieldProvider: PropTypes.any.apply,
    formOptions: PropTypes.any,
    name: PropTypes.string,
    label: PropTypes.string,
    description: PropTypes.string,
    isDanger: PropTypes.bool
};

DescriptiveCheckbox.defaultProps = {
    name: '',
    label: '',
    isDanger: false
};

export default DescriptiveCheckbox;
