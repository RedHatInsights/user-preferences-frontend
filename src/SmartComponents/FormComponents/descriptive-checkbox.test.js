import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import DescriptiveCheckbox from './DescriptiveCheckbox';
import { Form } from '@data-driven-forms/react-form-renderer';

describe('Descriptive checkbox tests', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <Form onSubmit={() => undefined}>
        {(props) => (
          <DescriptiveCheckbox label="test label" clearedValue {...props} />
        )}
      </Form>
    );
    expect(toJson(wrapper.find('DescriptiveCheckbox'))).toMatchSnapshot();
  });
});
