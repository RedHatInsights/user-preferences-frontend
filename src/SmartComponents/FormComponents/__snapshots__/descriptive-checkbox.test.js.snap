// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Descriptive checkbox tests should render correctly 1`] = `
<DescriptiveCheckbox
  isDanger={false}
  label="test label"
  name=""
>
  <Checkbox
    className="pref-c__descriptive-checkbox"
    description={
      <span
        className="pref-c__checkbox-description"
      />
    }
    id="descriptive-checkbox-"
    isChecked={false}
    isDisabled={false}
    isValid={true}
    label={
      <span
        className="pref-c__checkbox-label"
      >
        test label
      </span>
    }
    onChange={[Function]}
  >
    <div
      className="pf-c-check pref-c__descriptive-checkbox"
    >
      <input
        aria-invalid={false}
        className="pf-c-check__input"
        disabled={false}
        id="descriptive-checkbox-"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="pf-c-check__label"
        htmlFor="descriptive-checkbox-"
      >
        <span
          className="pref-c__checkbox-label"
        >
          test label
        </span>
      </label>
      <div
        className="pf-c-check__description"
      >
        <span
          className="pref-c__checkbox-description"
        />
      </div>
    </div>
  </Checkbox>
</DescriptiveCheckbox>
`;
